frame/election-provider-multi-phase/src/unsigned.rs

```diff
@@ -60,24 +65,62 @@ impl From<FeasibilityError> for MinerError {
 	}
 }
 
+/// Save a given call into OCW storage.
+fn save_solution<T: Config>(call: &Call<T>) {
+	let storage = StorageValueRef::persistent(&OFFCHAIN_CACHED_CALL);
+	storage.set(&call);
+}
+
+/// Get a saved solution from OCW storage if it exists.
+fn restore_solution<T: Config>() -> Option<Call<T>> {
+	StorageValueRef::persistent(&OFFCHAIN_CACHED_CALL).get().flatten()
+}
+
 impl<T: Config> Pallet<T> {
-	/// Mine a new solution, and submit it back to the chain as an unsigned transaction.
-	pub fn mine_check_and_submit() -> Result<(), MinerError> {
+	/// Attempt to restore a solution from cache. Otherwise, compute it fresh. Either way, submit.
+	pub fn restore_or_compute_then_submit() -> Result<(), MinerError> {
+		let call = match restore_solution() {
+			Some(call) => call,
+			None => {
+				let call = Self::mine_call()?;
+				save_solution(&call);
+				call
+			}
+		};
+		Self::submit_call(call)
```

Comment:
not big deal but note that if this `call` is coming from `Some(call)`, it could have an old `round`, and we're not checking it. But it will be caught down the road in `ValidateUnsigned`. agreed?

Reply:
That being said, it is easy enough to do filter it as well. I'd rewrite as: 

```
let call = restore_solution::<T>().and_then(|call| /* return `Some` only if `call` is recent */).or_else(|| { /* ompute a new call */ }); 
```

looks actually pretty clean to me.

Reply:
Yeah, good idea. 576f87cfe